bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -486,25 +527,29 @@ object BagsManager {
       }
     }
 
-    obs.getProgram.removeStructureChangeListener(StructureListener)
-    obs.getProgram.removeCompositeChangeListener(ChangeListener)
-    applySelection(TpeContext(obs))
-
-    // Update the TPE if it is visible
-    selOpt.foreach { sel =>
-      Option(TpeManager.get()).filter(_.isVisible).foreach { tpe =>
-        sel.assignments.foreach { ass =>
-          val clazz = ass.guideProbe.getType match {
-            case GuideProbe.Type.AOWFS => classOf[Altair_WFS_Feature]
-            case GuideProbe.Type.OIWFS => classOf[OIWFS_Feature]
-            case GuideProbe.Type.PWFS  => classOf[TpePWFSFeature]
+    def doApply(obs: ISPObservation): Unit = {
+      obs.getProgram.removeStructureChangeListener(StructureListener)
+      obs.getProgram.removeCompositeChangeListener(ChangeListener)
+      applySelection(TpeContext(obs))
```

Comment:
Same question here about racing. I guess we do this a lot in the OT and just assume all updates happen on the EDT.